Actions/DetermineDeploymentEnvironments/DetermineDeploymentEnvironments.ps1

```diff
@@ -139,18 +141,22 @@ else {
             "SyncMode" = $null
             "continuousDeployment" = $null
             "runs-on" = @($settings."runs-on".Split(',').Trim())
+            "shell" = $settings."shell"
         }
 
         # Check DeployTo<environmentName> setting
         $settingsName = "DeployTo$envName"
         if ($settings.ContainsKey($settingsName)) {
             # If a DeployTo<environmentName> setting exists - use values from this (over the defaults)
             $deployTo = $settings."$settingsName"
-            foreach($key in 'EnvironmentType','EnvironmentName','Branches','Projects','SyncMode','ContinuousDeployment','runs-on') {
+            foreach($key in 'EnvironmentType','EnvironmentName','Branches','Projects','SyncMode','ContinuousDeployment','runs-on','shell') {
                 if ($deployTo.ContainsKey($key)) {
                     $deploymentSettings."$key" = $deployTo."$key"
                 }
             }
+            if ($deploymentSettings."shell" -ne 'pwsh' -and $deploymentSettings."shell" -ne 'powershell') {
```

Comment:
Shouldn't this validation happen in ReadSettings rather? I guess shell should always be either pwsh or powershell regardless of whether you deploy to and environment or not

Reply:
Problem is really that these shell settings are under DeployToXXX - and we don't enumerate DeployTo's in ReadSettings.
Testing it here ensures that we don't use a different shell for these.